src/Geta.SEO.Sitemaps/Repositories/SitemapRepository.cs

```diff
@@ -48,7 +54,12 @@ public SitemapData GetSitemapData(string requestUrl)
             
             var host = url.Path.TrimStart('/').ToLowerInvariant();
 
-            return GetAllSitemapData().FirstOrDefault(x => GetHostWithLanguage(x) == host && (x.SiteUrl == null || x.SiteUrl.Contains(url.Host)));
+            var siteDefinition = _siteDefinitionResolver.GetByHostname(url.Host, true, out _);
+            if (siteDefinition == null)
+            {
+                return null;
+            }
+            return GetAllSitemapData().FirstOrDefault(x => GetHostWithLanguage(x) == host && (x.SiteUrl == null || siteDefinition.Hosts.Any(h => h.Name == new Url(x.SiteUrl).Host)));
```

Comment:
Can you split this up? It's hard to read and the line is too long

Reply:
Added unit tests so we easily test the functionality